addons/website/models/ir_ui_view.py

```diff
@@ -78,40 +68,75 @@ def _views_get(self, cr, uid, view_id, options=True, bundles=False, context=None
             xpath += "| //t[@t-call-assets]"
         for child in node.xpath(xpath):
             try:
-                called_view = self._view_obj(cr, uid, child.get('t-call', child.get('t-call-assets')), context=context)
+                called_view = self._view_obj(child.get('t-call', child.get('t-call-assets')))
             except ValueError:
                 continue
             if called_view not in result:
-                result += self._views_get(cr, uid, called_view, options=options, bundles=bundles, context=context)
+                result += self._views_get(called_view, options=options, bundles=bundles)
 
         extensions = view.inherit_children_ids
         if not options:
             # only active children
-            extensions = (v for v in view.inherit_children_ids if v.active)
+            extensions = view.inherit_children_ids.filtered('active')
 
         # Keep options in a deterministic order regardless of their applicability
         for extension in sorted(extensions, key=lambda v: v.id):
-            for r in self._views_get(
-                    cr, uid, extension,
+            for r in self._views_get(extension,
                     # only return optional grandchildren if this child is enabled
                     options=extension.active,
-                    context=context, root=False):
+                    root=False):
                 if r not in result:
                     result.append(r)
         return result
 
-    @tools.ormcache_context('uid', 'xml_id', keys=('website_id',))
-    def get_view_id(self, cr, uid, xml_id, context=None):
-        if context and 'website_id' in context and not isinstance(xml_id, (int, long)):
-            domain = [('key', '=', xml_id), '|', ('website_id', '=', context['website_id']), ('website_id', '=', False)]
-            [view_id] = self.search(cr, uid, domain, order='website_id', limit=1, context=context) or [None]
+    @api.model
+    @tools.ormcache_context('self.env.uid', 'xml_id', keys=('website_id',))
+    def get_view_id(self, xml_id):
+        if 'website_id' in self.env.context and not isinstance(xml_id, (int, long)):
+            domain = [('key', '=', xml_id), '|', ('website_id', '=', self.env.context['website_id']), ('website_id', '=', False)]
+            view_id = self.search(domain, order='website_id', limit=1).id or [None]
             if not view_id:
                 _logger.warning("Could not find view object with xml_id '%s'" % (xml_id))
-                raise ValueError('View %r in website %r not found' % (xml_id, context['website_id']))
+                raise ValueError('View %r in website %r not found' % (xml_id, self.env.context['website_id']))
         else:
-            view_id = super(view, self).get_view_id(cr, uid, xml_id, context=context)
+            view_id = super(View, self).get_view_id(xml_id)
         return view_id
 
+    # TODO - miration: website test case break
```

Comment:
`base/ir` has been recently migrated into new API. Maybe this will avoid using `api.v7` decorator ;)